pkg/cloud_provider/file/file.go

```diff
@@ -166,22 +166,34 @@ var (
 	shareUriRegex    = regexp.MustCompile(`^projects/([^/]+)/locations/([^/]+)/instances/([^/]+)/shares/([^/]+)$`)
 )
 
-func NewGCFSService(version string, client *http.Client, endpoint string) (Service, error) {
+func NewGCFSService(version string, client *http.Client, baseEndpoint string, multishareEndpoint string) (Service, error) {
 	ctx := context.Background()
 
-	fileService, err := filev1beta1.NewService(ctx, option.WithHTTPClient(client))
+	opts := []option.ClientOption{option.WithHTTPClient(client)}
+
+	var fsOpts []option.ClientOption
+	if baseEndpoint != "" {
+		fsOpts = append(opts, option.WithEndpoint(createFilestoreEndpointUrlBasePath(baseEndpoint)))
```

Comment:
as commented above we can use the same endpoint for both type of clients

Reply:
Done! Thanks for the context.